src/lib.rs

```diff
@@ -445,6 +452,33 @@ pub fn watchdog_enabled(unset_env: bool, usec: &mut u64) -> bool {
     }
 }
 
+fn monotonic_time_usecs() -> u128 {
+    use libc::{timespec, CLOCK_MONOTONIC, clock_gettime};
+
+    let mut monotonic_time = timespec {
+        tv_sec: 0,
+        tv_nsec: 0,
+    };
+
+    let result = unsafe {
+        clock_gettime(CLOCK_MONOTONIC, &mut monotonic_time)
+    };
+
+    if result != 0 {
+        // Should we return this as Result or Option?
+        return 0;
```

Comment:
Yeah, let's use https://doc.rust-lang.org/std/io/struct.Error.html#method.last_os_error here. I see that BIND 9 [aborts](https://gitlab.isc.org/isc-projects/bind9/-/blob/main/lib/isc/time.c#L118) when this fails.